docs: Revamp manage-tokens chapters (#777)
oasisprotocol/docs: docs/general/manage-tokens/README.mdx

protocol, in the governance proposal votes, the staking rewards are paid out in
the native token and the gas fees are covered by it.

## ROSE and the ParaTimes

The [Oasis network architecture] discriminates between the **consensus** and the
**compute** layer. The consensus and each ParaTime running on the compute layer
has their own **ledger** containing, among others, the **balances of the
accounts**.

Each ROSE token **can only exist in one ledger at once** (see [ADR-3] for
technical specifications). Moving tokens from the consensus layer to a ParaTime
is called a **deposit** and moving them from a ParaTime back to the consensus
layer is called a **withdrawal**. You can **transfer** tokens from your account
to another account only, if they both live on the consensus layer or inside the
same ParaTime.

![Deposits, withdrawals, transfers](../images/manage-tokens/transfer_deposit_withdrawal.svg)

[ADR-3]: ../../adrs/0003-consensus-runtime-token-transfer.md
[Oasis network architecture]: ../oasis-network/README.mdx

## Account Formats and Encryption Schemes

Transactions on the consensus layer must be signed by using the **ed25519

> encryption scheme**. The addresses on the consensus layer use the

Reviewer: These are all digital signature schemes so this should be fixed everywhere.

---
description: Transfer, deposit, withdraw and delegate your ROSE
---

import DocCard from '@theme/DocCard';
import DocCardList from '@theme/DocCardList';
import {findSidebarItem} from '@site/src/sidebarUtils';

# Manage your Tokens

The **native token** on Oasis Mainnet is called **ROSE**. The native token plays
a crucial security role in the proof-of-stake block validation protocol and in
participating on governance proposal votes. Staking rewards are paid out in the
native token and smart contract users pay gas fees with ROSE.

## ROSE and the ParaTimes

The [Oasis Network architecture] separates between the **consensus** and the
**compute** layer. The consensus layer and each ParaTime running on the compute
layer have their own **ledger** containing, among other data, the **balances of
the accounts**.

> You can only use your ROSE token **one place at a time**—either on a consensus

Reviewer: ```suggestion
You can only use your ROSE token **one place at a time**—either on the consensus
```

layer or inside an individual ParaTime. Moving tokens from the consensus layer

> to a ParaTime a **deposit** and moving them from a ParaTime back to

Reviewer: ```suggestion
to a ParaTime is a **deposit** and moving them from a ParaTime back to
```

> the consensus layer a **withdrawal** (see [ADR-3] for technical

Reviewer: ```suggestion
the consensus layer is a **withdrawal** (see [ADR-3] for technical
```

specifications). You can **transfer** tokens from your account to another

> account only, if they are either on the consensus layer or inside the same

Reviewer: ```suggestion
account only, if both accounts are either on the consensus layer or inside the same
```

**[Bech-32 encoding]** and you can recognize them by a typical `oasis1` prefix.

ParaTimes can implement arbitrary signature schemes and address encodings. For
example, since the Sapphire and Emerald ParaTimes are EVM-compatible, they
implement the **secp256k1** scheme and prefer the **hex-encoded**
addresses and private keys starting with `0x`.

The table below summarizes the current state of the address formats, signature
schemes and compatible wallets.

| Consensus or ParaTime | Address Format | Digital Signature Scheme    | Supported Wallets                                           |
|-----------------------|----------------|-----------------------------|-------------------------------------------------------------|
| Consensus             | `oasis1`       | ed25519                     | <ul><li>Oasis Wallet - Web</li><li>Oasis Wallet - Browser Extension</li><li>Oasis CLI</li></ul> |
| Sapphire              | `0x`, `oasis1` | secp256k1, ed25519, sr25519 | <ul><li>Metamask and other EVM-compatible wallets (transfers only)</li><li>Oasis Wallet - Browser Extension</li><li>Oasis Wallet - Web (deposits and withdrawals only)</li><li>Oasis CLI</li></ul> |

> | Cipher                | `oasis1`       | ed25519, sr25519            | <ul><li>Oasis CLI</li></ul> |

Reviewer: Cipher also supports the secp256k1 scheme.

---
description: Transfer, deposit, withdraw and delegate your ROSE
---

import DocCard from '@theme/DocCard';
import DocCardList from '@theme/DocCardList';
import {findSidebarItem} from '@site/src/sidebarUtils';

# Manage your Tokens

The **native token** on Oasis Mainnet is called **ROSE**. The native token plays

> a crucial role in the proof-of-stake block validation protocol, in the

Reviewer 2: ```suggestion
a crucial security role in the proof-of-stake block validation protocol, in the
```
or some security mention?

> governance proposal votes, the staking rewards are paid out in the native token

Reviewer 2: ```suggestion
participating on governance proposal votes. Staking rewards are paid out in the native token
```

> and the gas fees are covered by it.

Reviewer 2: ```suggestion
and smart contract users pay gas fees with ROSE.
```

## ROSE and the ParaTimes

> The [Oasis network architecture] discriminates between the **consensus** and the

Reviewer 2: ```suggestion
The [Oasis network architecture] separates the **consensus** and the
```

I also like "delinates," "demarcates," or other synonyms. I think "discriminates" can be a soft trigger for an American audience.

Author: > I think "discriminates" can be a soft trigger for an American audience.

Fair point.

**compute** layer. The consensus layer and each ParaTime running on the compute

> layer have their own **ledger** containing, among others, the **balances of the

Reviewer 2: ```suggestion
layer have their own **ledger** containing, among other data, the **balances of the
```

accounts**.

> Each ROSE token **can only exist in one ledger at once** (see [ADR-3] for

Reviewer: That is not actually true. Depositing a token into a ParaTime is two things:
* **Consensus layer:** transferring the token from the user's account into the ParaTime's account (each ParaTtime has its own account in the consensus layer).
* **ParaTime:** minting the token into the user's account.

And withdrawing is the reverse:
* **ParaTime:** burning the token from the users' account.
* **Consensus layer:** transferring the token from the ParaTime's account into the user's account.

Author: This is a technical implementation which should go maybe into https://github.com/oasisprotocol/oasis-core/blob/master/docs/consensus/services/staking.md?

This chapter is focused for first-timers learning about Oasis Architecture and moving the tokens around, different accounts, different chains, so I wouldn't explain this here. I'll reword this to sth like "You can only use your ROSE tokens one place at a time".

technical specifications). Moving tokens from the consensus layer to a ParaTime

> is called a **deposit** and moving them from a ParaTime back to the consensus

Reviewer 2: ```suggestion
is a **deposit** transaction and moving them from a ParaTime back to the consensus
```

or 

```suggestion
is a **deposit** and moving them from a ParaTime back to the consensus
```

layer is called a **withdrawal**. You can **transfer** tokens from your account
to another account only, if they are either on the consensus layer or inside the

> same ParaTime. Beside moving the tokens across layers and accounts, you can also

Reviewer 2: ```suggestion
same ParaTime. Besides moving the tokens across layers and accounts, you can also
```

**[delegate tokens]** to a validator and **earn passive income** as a reward.

![Deposits, withdrawals, transfers](../images/manage-tokens/transfer_deposit_withdrawal.svg)

[delegate tokens]: staking-and-delegating.md
[ADR-3]: ../../adrs/0003-consensus-runtime-token-transfer.md
[Oasis network architecture]: ../oasis-network/README.mdx

## The Wallets

To sign the token-related transactions such as transfers, deposits, withdrawals
and delegations described above, you need a **private key** tied to the
corresponding account. Your keys are stored in *[crypto wallets]*.

[crypto wallets]: https://en.wikipedia.org/wiki/Cryptocurrency_wallet

> ### Non-Custodial Oasis Wallets

Reviewer: We should mention [Oasis Safe](https://safe.oasis.io) somewhere for people who want to use a multisig on Sapphire?

Author: Oasis Safe doesn't support delegations yet right?

Reviewer: Not yet, unless you use it with a custom dApp. It will support them as soon as we have a dApp that supports them.

The Oasis team developed the following **non-custodial wallets** for you. This
means that the keys for managing the tokens are **stored on your device** such

> as the laptop or a mobile phone and **you are responsible to keep it safe**:

Reviewer 2: ```suggestion
as a laptop or a mobile phone and **you are responsible to keep it safe**:
```

- **[Oasis Wallet - Web]**: Runs as a web application in your web browser, the private keys are encrypted
  with a password and stored inside the Browser's local store.

- **[Oasis Wallet - Browser extension]**: Runs as an extension to your Chrome-based browser, the private keys are
  encrypted with a password and stored inside the Browser's encrypted store.

- **[Oasis CLI]**: Runs in a command line, suitable for automation, the private keys are
  encrypted by a password and stored inside your home folder.

:::caution

> For your own security and peace of mind, please only use the wallets that are

Reviewer 2: Should this section go just under "The Wallets?"

listed here. **Using unofficial wallets can result in the permanent loss of your
ROSE!**

:::

[Oasis Wallet - Web]: oasis-wallets/web.mdx
[Oasis Wallet - Browser extension]: oasis-wallets/browser-extension.mdx
[Oasis CLI]: cli/README.md

### MetaMask

[MetaMask] is probably the most-known crypto wallet. However, it is an
**EVM-compatible** wallet. This means **you can only use it to check the account
balances and sign transactions on Sapphire and Emerald chains**. You cannot use

> it, for example, to sign consensus layer transactions or perform deposits and

Reviewer 2: Great callout. Maybe we should emphasis the limitations.

```suggestion
it, for example, to sign consensus layer transactions or perform **deposits** and
```

withdrawals to and from ParaTimes.

You can add the Sapphire RPC endpoint by clicking on the "Add to MetaMask"
button next to your preferred Mainnet endpoint provider in the [Sapphire]
chapter. Similarly, you will find a list of RPC endpoints for Emerald on the
[Emerald] page.

![Metamask - Adding Sapphire Mainnet Network Configuration](../images/wallet/metamask/settings.png)

[MetaMask]: https://metamask.io/download/
[Sapphire]: ../../dapp/sapphire/README.mdx#rpc-endpoints
[Emerald]: ../../dapp/emerald/README.mdx#rpc-endpoints

### Ledger

The wallets above are just carefully programmed computer programs that store
your keys (in an encrypted form) somewhere on your disk and then use them to

> sign the transactions. However, if your device gets infected by a malicious software

Reviewer 2: ```suggestion
sign the transactions. However, if your device gets infected with a piece of malicious software
```

schemes and compatible wallets.

| Consensus or ParaTime | Address Format     | Digital Signature Scheme           | Supported Wallets                                                |
|-----------|--------------------|-----------------------------|------------------------------------------------------------------|
| Consensus | `oasis1`           | ed25519                     | <ul><li>Oasis Wallet - Web</li><li>Oasis Wallet - Browser Extension</li><li>Oasis CLI</li></ul> |
| Sapphire  | `0x`, `oasis1` | secp256k1, ed25519, sr25519 | <ul><li>Metamask and other EVM-compatible wallets (transfers only)</li><li>Oasis Wallet - Browser Extension</li><li>Oasis Wallet - Web (deposits and withdrawals only)</li><li>Oasis CLI</li></ul> |
| Cipher    | `oasis1`           | ed25519, sr25519            | <ul><li>Oasis CLI</li></ul> |
| Emerald   | `0x`, `oasis1` | secp256k1, ed25519, sr25519 | <ul><li>Metamask and other EVM-compatible wallets (transfers only)</li><li>Oasis Wallet - Browser Extension</li><li>Oasis Wallet - Web (deposits and withdrawals only)</li><li>Oasis CLI</li></ul> |

[Bech-32 encoding]: https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32

> ## Obtain ROSE

Reviewer 2: ```suggestion
## Obtaining ROSE
```

The most common way to obtain ROSE is by buying it on a centralized
[cryptocurrency exchange]. At time of writing this chapter, all exchanges
operated solely on the **consensus layer**. This means that you can only move
the tokens in or out of the exchange using **your consensus account**.

An alternative way to obtain ROSE is by swapping another cryptocurrency with

> ROSE on an on-chain exchange also known as **the [decentralized exchange] (DEX)**.

Reviewer: ```suggestion
ROSE on an on-chain exchange also known as **a [decentralized exchange] (DEX)**.
```

> At time of writing at least one such exchange was operating on Sapphire. In this

Reviewer 2: ```suggestion
At time of writing at least one such exchange is operating on Sapphire. In this
```

The "Amount" is a sum of three values:

- the "Available" tokens that can immediately be transferred,
- the "Escrow" tokens that are delegated,
- the "Reclaim" tokens that are waiting for the debonding period to pass.

To check the account's deposits and withdrawals navigate to "Transactions" pane
and press "ParaTime" on the right side, next to the "Consensus" button.
You will see all ParaTime-related transactions including deposits, withdrawals,
transfers and even smart contract transactions.

![Search result of oasis1 address - Account details](../images/manage-tokens/oasisscan_paratime_txes.png)

Furthermore, you can view the transaction details, if you click on a
transaction's "Tx Hash". Among others, you will see the transaction type, the
"from", "to" and "amount" fields.

![Tx Hash - Transaction details](../images/manage-tokens/oasisscan_paratime_tx_details.png)

:::info

Be aware that the [Oasis Scan Blockchain Explorer](https://www.oasisscan.com)
is built for consensus layer. If you want to explore Sapphire (0x addresses,
Token Transfers, Contract Calls, etc.), you have to use the
[Sapphire Blockchain Explorer](https://explorer.oasis.io/mainnet/sapphire).

:::

> ## See Also

Reviewer 2: (maybe)
```suggestion
## Additional Resources
```

Author: Actually the whole See Also section can be removed, because those topics were already covered inside the chapter. The See Also usually refers to similar chapters in other parts (e.g. from the core or node operators).
